### PR TITLE
ci: Add testing on macOS

### LINKED ISSUE #157: Run tests on macOS
For declaring macOS support, we need to run tests on macOS. In this issue, we need to extend test environments in Github Actions with macOS.

Don't forget to add macOS as supported platform in th README.

### PR DESCRIPTION
**Pull request by better0fdead**
Added macOS testing into ci.
- [not needed] Tests
- [not needed] Changelog
- [not needed] Documentation 

Closes #157

### CODE REVIEW

File: .github/workflows/testing.yml [hunk 1]
```diff
@@ -195,3 +197,179 @@ jobs:
...
+      (github.event_name == 'workflow_dispatch')
+
+    strategy:
+      fail-fast: false
+      matrix:
+        golang:
+          - 1.13
+        runs-on:
+          - macos-10.15
+          - macos-11.0
```
Comment (oleg-jukovec) [inline .github/workflows/testing.yml:219]: Here we can limit to one macos version (a latest one?)/

File: .github/workflows/testing.yml [hunk 2]
```diff
@@ -126,14 +126,16 @@ jobs:
...
-        # if `ref` and `repository` are empty.
+        if: github.event_name == 'pull_request_target'
         with:
-          ref: ${{github.event.pull_request.head.ref}}
-          repository: ${{github.event.pull_request.head.repo.full_name}}
+          ref: refs/pull/${{ github.event.pull_request.number }}/merge
+
+      - name: Clone the connector
+        if: github.event_name != 'pull_request_target'
+        uses: actions/checkout@v2
```
Comment (oleg-jukovec) [inline .github/workflows/testing.yml:138]: The change looks like a rebase artifact.

File: .github/workflows/testing.yml [hunk 3]
```diff
@@ -195,3 +197,179 @@ jobs:
...
+      matrix:
+        golang:
+          - 1.13
+        runs-on:
+          - macos-10.15
+          - macos-11.0
+        tarantool:
+          - brew
+          - 1.10.10
+          - 2.8.1
```
Comment (oleg-jukovec) [inline .github/workflows/testing.yml:223]: Is there no way to use pre-build packages on MacOS?

**Review by oleg-jukovec — CHANGES_REQUESTED** [review]
Thank you for the patch, but we need to fix tests before merge. Could you do it?

`ci: Add testing on macOS` -> `ci: add testing on macOS`.

File: .github/workflows/testing.yml (hunk 1 above)
Comment (Totktonada) [reply .github/workflows/testing.yml:219]: Moreover, 10.15 is going to EOL in GitHub Actions. And `macos-12` (not `macos-12.0`!) appears now. Please, catch recent updates from https://github.com/tarantool/smtp/commits/master/.github/workflows/testing.yml.

File: .github/workflows/testing.yml (hunk 3 above)
Comment (Totktonada) [reply .github/workflows/testing.yml:223]: I'm thinking on it in background. Of course, it is possible and there are different ways with cons and pros. We should setup some infrastructure and support it in setup-tarantool. Reach me if you want to participate and/or push this activity.

File: .github/workflows/testing.yml [hunk 4]
```diff
@@ -195,3 +197,179 @@ jobs:
...
+          # See https://github.com/github/docs/issues/14145
+          key: ${{ matrix.runs-on }}-${{ matrix.tarantool }}-v4
+        if: matrix.tarantool != 'brew' && matrix.tarantool != 'master'
+
+      - name: Install tarantool build dependencies
+        run: brew install autoconf automake libtool openssl@1.1
+        if: matrix.tarantool != 'brew' && steps.cache.outputs.cache-hit != 'true'
+
+      - name: Clone tarantool ${{ env.T_VERSION }}
+        uses: actions/checkout@v2
```
Comment (DifferentialOrange) [inline .github/workflows/testing.yml:265]: checkout@v2 uses node 12 which will be deprecated soon. It's better to migrate to v3 already.

File: .github/workflows/testing.yml (hunk 3 above)
Comment (DifferentialOrange) [reply .github/workflows/testing.yml:223]: Yeah, building Tarantool from sources for module CI seems painful. I think we must get rid of it if possible.

**Review by DifferentialOrange — COMMENTED** [review]
Does the whole "source install" thing was inspired by some existing CI/CD pipeline?

File: .github/workflows/testing.yml (hunk 1 above)
Comment (better0fdead) [reply .github/workflows/testing.yml]: According to tarantool.io tarantool supports 11, 12 versions of macOS. Changed according to that information.

File: .github/workflows/testing.yml (hunk 4 above)
Comment (better0fdead) [reply .github/workflows/testing.yml]: Added commit to update actions.

**Comment by better0fdead** [comment]
> Thank you for the patch, but we need to fix tests before merge. Could you do it?
> 
> `ci: Add testing on macOS` -> `ci: add testing on macOS`.

I suggest someone to take a look into it, since I am not very familiar with queue and it will probably take many time. Also I will be busy doing other tasks. If time is not very important I will try to fix.

File: .github/workflows/testing.yml (hunk 3 above)
Comment (better0fdead) [reply .github/workflows/testing.yml:223]: Maybe cache will resolve the problem.  You can check it [here](https://github.com/tarantool/go-tarantool/actions/runs/3584174473/jobs/6030512155) macOS 12 Tarantool 2.8.1 test.

File: queue/queue_test.go [hunk 5]
```diff
@@ -782,7 +782,7 @@ func TestUtube_Put(t *testing.T) {
 			return
 		}
 
-		time.Sleep(2 * time.Second)
+		time.Sleep(2*time.Second + 30*time.Millisecond)
```
Comment (DifferentialOrange) [inline queue/queue_test.go:785]: Such approach is definitely not good.

Comment (oleg-jukovec) [reply queue/queue_test.go:785]: I'll try to explain:

A situation that is solved by increasing the waiting time by a bit looks like a data race. Such a fix should be done only if you are absolutely sure that this is the expected behavior, and not a data race (and it is advisable to add a comment on this).

Comment (better0fdead) [reply queue/queue_test.go:785]: I understand it, it was done for experimentational purposes on github runners. I will not use it or give a full explanation for it.

**Pull request closed without merging**